app/mm-redux/utils/theme_utils.ts

```diff
@@ -96,21 +96,76 @@ export function blendColors(background: string, foreground: string, opacity: num
         opacity,
     );
 
+    if (hex) {
+        let r = red.toString(16);
+        let g = green.toString(16);
+        let b = blue.toString(16);
+
+        if (r.length === 1) {
+            r = '0' + r;
+        }
+        if (g.length === 1) {
+            g = '0' + g;
+        }
+        if (b.length === 1) {
+            b = '0' + b;
+        }
+
+        return `#${r + g + b}`;
+    }
+
     return `rgba(${red},${green},${blue},${alpha})`;
 }
 
+// object mapping theme types to their respective keys for retrieving the source themes directly
+// - supports mapping old themes to new themes
+const themeTypeMap: ThemeTypeMap = {
+    Mattermost: 'denim',
+    Organization: 'sapphire',
+    'Mattermost Dark': 'indigo',
+    'Windows Dark': 'onyx',
+    Denim: 'denim',
+    Sapphire: 'sapphire',
+    Quartz: 'quartz',
+    Indigo: 'indigo',
+    Onyx: 'onyx',
+};
+
 // setThemeDefaults will set defaults on the theme for any unset properties.
-export function setThemeDefaults(theme: Theme): Theme {
-    const defaultTheme = Preferences.THEMES.default;
+export function setThemeDefaults(theme: Partial<Theme>): Theme {
```

Comment:
is this function used anywhere? I can't seem to find any reference to it

Reply:
It is used in `app/mm-redux/selectors/entities/preferences.ts` line 151. It wasn't previously, but I updated to match the implementation of the Webapp for consistency. The Webapp uses it in more places so having the separate function is helpful. Given mobile only uses it once, maybe it's not necessary to keep it separate? 0/5 either way. Thoughts @enahum?

Reply:
0/5 if we want to keep the function..

What I can tell you is that this is no longer used in gekidou, as it was part of a redux selector, do you think what this function does is important to port as well ? perhaps this could be part of the `makeStyleSheetFromTheme` instead?

Reply:
Got it, well for Gekidou this is not being used at all, in case if needed while we develop Gekidou we could add it in.

Thanks for the explanation